File: Service/CustomerService.cs
using System.Collections.Generic;
using System.Linq;
using DTO;

namespace Service
{
    public class CustomerService: ICustomerService
    {
        private readonly IBetService _betService;

        public CustomerService(IBetService betService)
        {
            _betService = betService;
        }

        public IList<Customer> GetAll()
        {
            // customers retrieved from bet data
            var custIds = _betService.GetAll().Select(bet => bet.CustomerId).Distinct();

            return (from bet in custIds
                                         select new Customer
                                         {
                                             Id = bet
                                         }).Distinct().ToList();
        }
    }
}

File: Risky/Scripts/Src/services/customer-service.js
import qwest from 'qwest';

/*
 * Customer service for retrieving customer data.
 *
 */
function getCustomers() {
    return qwest.get('api/customers/')
        .then((xhr, response) => response);
}

// customer service
export default {
    getAll: getCustomers
};
